File: samples/server/petstore/kotlin-springboot/src/main/kotlin/org/openapitools/api/PetApiService.kt
package org.openapitools.api

import org.openapitools.model.ModelApiResponse
import org.openapitools.model.Pet

interface PetApiService {

    fun addPet(body: Pet): Unit

    fun deletePet(petId: Long, apiKey: String?): Unit

    fun findPetsByStatus(status: List<String>): List<Pet>

    fun findPetsByTags(tags: List<String>): List<Pet>

    fun getPetById(petId: Long): Pet

    fun updatePet(body: Pet): Unit

    fun updatePetWithForm(petId: Long, name: String?, status: String?): Unit

    fun uploadFile(petId: Long, additionalMetadata: String?, file: org.springframework.core.io.Resource?): ModelApiResponse
}

File: samples/openapi3/client/petstore/typescript/builds/inversify/apis/PetApi.service.ts
import type { Configuration } from "../configuration";
import type { HttpFile, RequestContext, ResponseContext, HttpInfo } from "../http/http";

import { ApiResponse } from "../models/ApiResponse";
import { Pet } from "../models/Pet";

export abstract class AbstractPetApiRequestFactory {
    public abstract addPet(pet: Pet, options?: Configuration): Promise<RequestContext>;

    public abstract deletePet(petId: number, apiKey?: string, options?: Configuration): Promise<RequestContext>;

    public abstract findPetsByStatus(status: Array<'available' | 'pending' | 'sold'>, options?: Configuration): Promise<RequestContext>;

    public abstract findPetsByTags(tags: Array<string>, options?: Configuration): Promise<RequestContext>;

    public abstract getPetById(petId: number, options?: Configuration): Promise<RequestContext>;

    public abstract updatePet(pet: Pet, options?: Configuration): Promise<RequestContext>;

    public abstract updatePetWithForm(petId: number, name?: string, status?: string, options?: Configuration): Promise<RequestContext>;

    public abstract uploadFile(petId: number, additionalMetadata?: string, file?: HttpFile, options?: Configuration): Promise<RequestContext>;

}


export abstract class AbstractPetApiResponseProcessor {
     public abstract addPetWithHttpInfo(response: ResponseContext): Promise<HttpInfo<Pet >>;

     public abstract deletePetWithHttpInfo(response: ResponseContext): Promise<HttpInfo< void>>;

     public abstract findPetsByStatusWithHttpInfo(response: ResponseContext): Promise<HttpInfo<Array<Pet> >>;

     public abstract findPetsByTagsWithHttpInfo(response: ResponseContext): Promise<HttpInfo<Array<Pet> >>;

     public abstract getPetByIdWithHttpInfo(response: ResponseContext): Promise<HttpInfo<Pet >>;

     public abstract updatePetWithHttpInfo(response: ResponseContext): Promise<HttpInfo<Pet >>;

     public abstract updatePetWithFormWithHttpInfo(response: ResponseContext): Promise<HttpInfo< void>>;

     public abstract uploadFileWithHttpInfo(response: ResponseContext): Promise<HttpInfo<ApiResponse >>;

}
